plugins/aws/resoto_plugin_aws/resource/apigateway.py

```diff
@@ -0,0 +1,447 @@
+    }
+    types: List[str] = field(factory=list)
+    vpc_endpoint_ids: List[str] = field(factory=list)
+
+
+@define(eq=False, slots=False)
+class AwsApiGatewayRestApi(ApiGatewayTaggable, AwsResource):
+    kind: ClassVar[str] = "aws_api_gateway_rest_api"
+    api_spec: ClassVar[AwsApiSpec] = AwsApiSpec("apigateway", "get-rest-apis", "items")
+    reference_kinds: ClassVar[ModelReference] = {
+        "successors": {
+            "default": [
+                "aws_vpc_endpoint",
+                "aws_api_gateway_deployment",
+                "aws_api_gateway_authorizer",
+                "aws_api_gateway_resource",
+            ],
+            "delete": ["aws_vpc_endpoint"],
+        }
+    }
+
+    @classmethod
+    def called_apis(cls) -> List[AwsApiSpec]:
+        return [
+            cls.api_spec,
+            AwsApiSpec("apigateway", "get-deployments"),
+            AwsApiSpec("apigateway", "get-stages"),
+            AwsApiSpec("apigateway", "get-authorizers"),
+            AwsApiSpec("apigateway", "get-resources"),
+        ]
+
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "id": S("id"),
+        "name": S("name"),
+        "tags": S("tags", default=[]),
+        "ctime": S("createdDate"),
+        "description": S("description"),
+        "api_version": S("version"),
+        "api_warnings": S("warnings", default=[]),
+        "api_binary_media_types": S("binaryMediaTypes", default=[]),
+        "api_minimum_compression_size": S("minimumCompressionSize"),
+        "api_key_source": S("apiKeySource"),
+        "api_endpoint_configuration": S("endpointConfiguration") >> Bend(AwsApiGatewayEndpointConfiguration.mapping),
+        "api_policy": S("policy"),
+        "api_disable_execute_api_endpoint": S("disableExecuteApiEndpoint"),
+    }
+    description: Optional[str] = field(default=None)
+    api_version: Optional[str] = field(default=None)
+    api_warnings: List[str] = field(factory=list)
+    api_binary_media_types: List[str] = field(factory=list)
+    api_minimum_compression_size: Optional[int] = field(default=None)
+    api_key_source: Optional[str] = field(default=None)
+    api_endpoint_configuration: Optional[AwsApiGatewayEndpointConfiguration] = field(default=None)
+    api_policy: Optional[str] = field(default=None)
+    api_disable_execute_api_endpoint: Optional[bool] = field(default=None)
+
+    @classmethod
+    def collect(cls: Type[AwsResource], json: List[Json], builder: GraphBuilder) -> None:
+        for js in json:
+            api_instance = cls.from_api(js)
+            region = builder.region
+            api_instance.arn = f"arn:{arn_partition(region)}:apigateway:{region.id}::/restapis/{api_instance.id}"
+            builder.add_node(api_instance, js)
+            for deployment in builder.client.list("apigateway", "get-deployments", "items", restApiId=api_instance.id):
+                deploy_instance = AwsApiGatewayDeployment.from_api(deployment)
```

Comment:
Consider to add a comment in AwsApiGatewayDeployment, that this resource is collected via AwsApiGatewayRestApi. Same for other resources collected here.